src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,50 @@
 package core.basesyntax.service;
 
+import core.basesyntax.RegistrationFailedException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 130;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkUser(user);
+        return storageDao.add(user);
+    }
+
+    private void checkUser(User user) {
+        if (user == null) {
+            throw new RegistrationFailedException("User can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationFailedException("User age can't be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationFailedException("User login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationFailedException("User password can't be null");
+        }
+        if (user.getLogin().length() == 0) {
+            throw new RegistrationFailedException("User login must be longer than 0 characters");
```

Comment:
Magic number should be constant. Will your message be correct if you change min length on the 2?